### ISSUE TITLE
Phase 1 Roadmap

### ISSUE THREAD

**Issue by Dar-rius** [Dar-rius/zeroRL#43]
Hi everyone, 

Our first release [0.1.1](https://github.com/Dar-rius/zeroRL/releases/tag/0.1.1) is out!

For this release, we focused on designing a simple, modular RL research framework architecture with minimal abstractions, while providing the core features needed to prototype new algorithms:

* **A modular training orchestrator** including rollout collection, visualization (W&B, TensorBoard), observation normalization, customizable weight update functions, model checkpointing, and profiling.
* **Data structure classes** to easily configure training runs and RL algorithms.
* **Abstraction classes** for designing agents and environments, providing a safer interface without restricting customization.
* **An extensible tensor buffer** capable of storing multiple variables.
* **PPO as the first algorithm** included in the framework, with a modular design allowing researchers to modify individual parts of the algorithm according to their needs.

Our next steps are to expand algorithm support and prove the framework's reliability.

***This roadmap is a living document and will naturally evolve based on feedback and the needs of the research community.***

## Current Focus

Features we are actively working on right now.

- [ ] Create a documentation website.
- [ ] Add an cnn architecture to the ActorCriticAgent factory.

## Planned Features

Next in the pipeline once the current focus is stable.

### Core

- [ ] Add system of hyperparam sweeps for hyperparams research easier.
- [ ] Allow enginers/researchers to experiments some multiagent systems and implement MAPPO.
- [ ] Save the NormMeanStd state for simulators such as Mujoco and Isaac sim.
- [x] Add necessary buffer and agent architectures to support algorithms that do not use an actor-critic architecture.
- [x] Automatically resize the buffer
- [x] Add tools to help engineers and researchers understand what is happening inside their agents, buffers, training loop and other.

### Documentation

- [ ] Add more examples based on research use cases.
  
### Implement basic RL algorithms

  - [ ] SAC
  - [ ] DQN 
  - [ ] DDPG
  - [ ] Recurrent PPO
  

### Performance and stability

- [ ] Add tests to verify that RL algorithm implementations are correct, robust and performant.
- [ ] Add benchmarks to evaluate sample efficiency.

### Compatibility and portability

- [ ] Add  support for Metal-compatible PyTorch backend, check the [issue](https://github.com/Dar-rius/zeroRL/issues/59)
- [ ] Reduce the number of dependencies
- [ ] Support for most environments
